---
title: "PeakCalling"
author: "Yingying Cao"
date: "2/26/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

From the paper Integrative analysis of 111 reference human epigenomes:

Peak calling. For the histone ChIP-seq data, the MACSv2.0.10 peak caller was used to compare ChIP-seq signal to a corresponding whole-cell extract (WCE) sequenced control to identify narrow regions of enrichment (peaks) that pass a Poisson P value threshold 0.01, broad domains that pass a broad-peak Poisson P value of 0.1 and gapped peaks which are broad domains (P < 0.1) that include at least one narrow peak (P < 0.01) (https://github.com/taoliu/MACS/). Fragment lengths for each data set were pre-estimated using strand cross-correlation analysis and the SPP peak caller package and these fragment length estimates were explicitly used as parameters in the MACS2 program (â€“shift-size = fragment_length/2).

### merge technical and/or biological replicates
Merging BAM files of technical and/or biological replicates can improve the sensitivity of the peak calling by increasing the depth of read coverage. 

```{bash}
# wild type H3K27ac
samtools merge ./data/tet2/output/merged/wildtypeH3K27ac.bam ./data/tet2/output/SRR925642_sorted.bam ./data/tet2/output/SRR925643_sorted.bam ./data/tet2/output/SRR925653_sorted.bam

# tet2 out H3K27ac
samtools merge ./data/tet2/output/merged/tet2outH3K27ac.bam ./data/tet2/output/SRR925675_sorted.bam ./data/tet2/output/SRR925682_sorted.bam

# wild type H3K27me3
samtools merge ./data/tet2/output/merged/wildtypeH3K27me3.bam ./data/tet2/output/SRR925644_sorted.bam ./data/tet2/output/SRR925645_sorted.bam ./data/tet2/output/SRR925646_sorted.bam ./data/tet2/output/SRR925654_sorted.bam

# tet2 out H3K27me3
samtools merge ./data/tet2/output/merged/tet2outH3K27me3.bam ./data/tet2/output/SRR925676_sorted.bam ./data/tet2/output/SRR925683_sorted.bam

# wild type H3K36me3
samtools merge ./data/tet2/output/merged/wildtypeH3K36me3.bam ./data/tet2/output/SRR925647_sorted.bam ./data/tet2/output/SRR925655_sorted.bam

# tet2 out H3K36me3
samtools merge ./data/tet2/output/merged/tet2outH3K36me3.bam ./data/tet2/output/SRR925677_sorted.bam ./data/tet2/output/SRR925684_sorted.bam

# wild type H3K4me1
samtools merge ./data/tet2/output/merged/wildtypeH3K4me1.bam ./data/tet2/output/SRR925639_sorted.bam ./data/tet2/output/SRR925650_sorted.bam

# tet2 out H3K4me1
samtools merge ./data/tet2/output/merged/tet2outH3K4me1.bam ./data/tet2/output/SRR925673_sorted.bam ./data/tet2/output/SRR925679_sorted.bam

# wild type H3K4me3
samtools merge ./data/tet2/output/merged/wildtypeH3K4me3.bam ./data/tet2/output/SRR925640_sorted.bam ./data/tet2/output/SRR925651_sorted.bam

# tet2 out H3K4me3
samtools merge ./data/tet2/output/merged/tet2outH3K4me3.bam ./data/tet2/output/SRR925674_sorted.bam ./data/tet2/output/SRR925680_sorted.bam

# wild type H3K9me3
samtools merge ./data/tet2/output/merged/wildtypeH3K9me3.bam ./data/tet2/output/SRR925641_sorted.bam ./data/tet2/output/SRR925652_sorted.bam

# tet2 out H3K9me3
samtools merge ./data/tet2/output/merged/tet2outH3K9me3.bam ./data/tet2/output/SRR925681_sorted.bam


# wild type input
samtools merge ./data/tet2/output/merged/wildtypeinput.bam ./data/tet2/output/SRR925648_sorted.bam ./data/tet2/output/SRR925649_sorted.bam ./data/tet2/output/SRR925656_sorted.bam

# tet2 out input
samtools merge ./data/tet2/output/merged/tet2outinput.bam ./data/tet2/output/SRR925678_sorted.bam ./data/tet2/output/SRR925685_sorted.bam
cd ./data/tet2/output/merged/
for i in *.bam; do samtools index $i; done


```




## estimate mean fragment length 
citation: https://code.google.com/archive/p/phantompeakqualtools/

It is generally set to --nomodel for histone marks and ATAC, not for TF.

The macs2 requires a prior width to extend the reads to the fixed fragment size (--extsize) when --nomodel is set.

This will be set to a number that was estimated by phantompeaktools (run_SPP.r)

This will not have a very big effect for broadpeaks. The default parameter, I think it is 200 and it's should be fine in general.

Running SPP not only gives you the estimated fragment size, but also the NSC and RSC values, which tells the quality of chips. So people tend to run SPP to know the quality of chip and use the estimated fragment size while calling peaks using macs.

P.S: The run_SPP.r sometimes returns multiple estimated fragments lengths and the highest one or average is taken.
```{r}
# download from
# https://github.com/crazyhottommy/phantompeakqualtools
devtools::install_github('hms-dbmi/spp', build_vignettes = FALSE)
library(caTools)
```

```{bash}
for i in ./data/tet2/output/merged/*.bam; do Rscript ./data/phantompeakqualtools/run_spp.R -c=$i -savp -out=$i.txt; done
```

### call peaks with macs2
for detailed explaination about the estimated fragment length from cross-correlation using in --extsize see https://github.com/crazyhottommy/ChIP-seq-analysis/blob/master/part0_quality_control.md#calculate-fragment-length-nsc-and-rsc-by-phantompeakqualtools
```{bash}

macs2 callpeak -t ./data/tet2/output/merged/wildtypeH3K27ac.bam -c ./data/tet2/output/merged/wildtypeinput.bam --broad -g mm --broad-cutoff 0.1 -n wildtypeH3K27ac --outdir ./data/tet2/macs2  --nomodel --extsize 165 2> ./data/tet2/macs2/wildtypeH3K27ac-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/tet2outH3K27ac.bam -c ./data/tet2/output/merged/tet2outinput.bam --broad -g mm --broad-cutoff 0.1 -n tet2outH3K27ac --outdir ./data/tet2/macs2  --nomodel --extsize 165 2> ./data/tet2/macs2/tet2outH3K27ac-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/wildtypeH3K27me3.bam -c ./data/tet2/output/merged/wildtypeinput.bam --broad -g mm --broad-cutoff 0.1 -n wildtypeH3K27me3 --outdir ./data/tet2/macs2  --nomodel --extsize 165 2> ./data/tet2/macs2/wildtypeH3K27me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/tet2outH3K27me3.bam -c ./data/tet2/output/merged/tet2outinput.bam --broad -g mm --broad-cutoff 0.1 -n tet2outH3K27me3 --outdir ./data/tet2/macs2  --nomodel --extsize 165 2> ./data/tet2/macs2/tet2outH3K27me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/wildtypeH3K36me3.bam -c ./data/tet2/output/merged/wildtypeinput.bam --broad -g mm --broad-cutoff 0.1 -n wildtypeH3K36me3 --outdir ./data/tet2/macs2  --nomodel --extsize 165 2> ./data/tet2/macs2/wildtypeH3K36me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/tet2outH3K36me3.bam -c ./data/tet2/output/merged/tet2outinput.bam --broad -g mm --broad-cutoff 0.1 -n tet2outH3K36me3 --outdir ./data/tet2/macs2  --nomodel --extsize 165 2> ./data/tet2/macs2/tet2outH3K36me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/wildtypeH3K4me1.bam -c ./data/tet2/output/merged/wildtypeinput.bam --broad -g mm --broad-cutoff 0.1 -n wildtypeH3K4me1 --outdir ./data/tet2/macs2  --nomodel --extsize 130 2> ./data/tet2/macs2/wildtypeH3K4me1-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/tet2outH3K4me1.bam -c ./data/tet2/output/merged/tet2outinput.bam --broad -g mm --broad-cutoff 0.1 -n tet2outH3K4me1 --outdir ./data/tet2/macs2  --nomodel --extsize 160 2> ./data/tet2/macs2/tet2outH3K4me1-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/wildtypeH3K4me3.bam -c ./data/tet2/output/merged/wildtypeinput.bam --broad -g mm --broad-cutoff 0.1 -n wildtypeH3K4me3 --outdir ./data/tet2/macs2  --nomodel --extsize 135 2> ./data/tet2/macs2/wildtypeH3K4me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/tet2outH3K4me3.bam -c ./data/tet2/output/merged/tet2outinput.bam --broad -g mm --broad-cutoff 0.1 -n tet2outH3K4me3 --outdir ./data/tet2/macs2  --nomodel --extsize 160 2> ./data/tet2/macs2/tet2outH3K4me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/wildtypeH3K9me3.bam -c ./data/tet2/output/merged/wildtypeinput.bam --broad -g mm --broad-cutoff 0.1 -n wildtypeH3K9me3 --outdir ./data/tet2/macs2  --nomodel --extsize 135 2> ./data/tet2/macs2/wildtypeH3K9me3-macs2.log

macs2 callpeak -t ./data/tet2/output/merged/tet2outH3K9me3.bam -c ./data/tet2/output/merged/tet2outinput.bam --broad -g mm --broad-cutoff 0.1 -n tet2outH3K9me3 --outdir ./data/tet2/macs2  --nomodel --extsize 135 2> ./data/tet2/macs2/tet2outH3K9me3-macs2.log


```
http://ginolhac.github.io/chip-seq/peak/
.broadPreak file
This file has the different fields:

    chromosome
    start
    end
    peak name
    integer score for display
    strand
    fold-change
    -log10 pvalue
    -log10 qvalue
    relative summit position to peak start
"fold enrichment for this peak summit against random Poisson distribution with local lambda, -log10(qvalue) at peak summit"

GappedPeak is a representation of narrow peaks as blocks over a broad peak. To trick the visualisation tools, they use the same format as gene models, but use the narrow peak coordinates as exons coordinates and the broad peak coordinates as coding region coordinate

```{r}
tet2outH3K27ac <- read.table("./data/tet2/macs2/tet2outH3K27ac_peaks.broadPeak")
mm.tss <- read.table("./data/tss.bed")
# cut -f 1-8 mm.bed > mm1-8.bed
mm <- read.table("./data/mm1-8.bed")
write.table(tet2outH3K27ac[,1:3], file="./data/tet2/chromhmm/tet2outH3K27acPeak.bed", sep="\t", quote=F, row.names=F, col.names=F)
```



